galsim/wfirst/__init__.py

```diff
@@ -0,0 +1,188 @@
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""
+The galsim.wfirst module, containing information GalSim needs to simulate images for the WFIRST-AFTA
+project.
+
+This module contains numbers and routines for the WFIRST-AFTA project.  There is also a demo
+illustrating the use of most of this functionality, demo13.py.  Some of the parameters below relate
+to the entire wide-field imager.  Others, especially the return values of the functions to get the
+PSF and WCS, are specific to each SCA (Sensor Chip Assembly, the equivalent of a chip for an optical
+CCD) and therefore are indexed based on the SCA.  All SCA-related arrays are 1-indexed, i.e., the
+entry with index 0 is None and the entries from 1 to n_sca are the relevant ones.  This is
+consistent with diagrams and so on provided by the WFIRST project, which are 1-indexed.
+
+Currently, the module includes the following numbers:
+
+    gain - The gain for all SCAs (sensor chip assemblies) is expected to be the same, so this is a
+           single value rather than a list of values.
+
+    pixel_scale - The pixel scale in units of arcsec/pixel.
+
+    diameter - The telescope diameter in meters.
+
+    obscuration - The linear obscuration of the telescope, in terms of fraction of the diameter.
+
+    exptime - The typical exposure time in units of seconds.  The number that is stored is for a
+              single dither.  Each location within the survey will be observed with a total of 5-7
+              dithers across 2 epochs.  For a single dither, there are 32 up-the-ramp samples, each
+              taking 5.423 seconds, but the effective live time is really 31 samples.  This is the
+              source of the 168.1s that is currently stored for exptime.
+
+    n_dithers - The number of dithers per filter (typically 5-7, so this is currently 6 as a
+                reasonable effective average).
+
+    dark_current - The dark current in units of e-/pix/s.
+
+    nonlinearity_beta - The coefficient of the (counts)^2 term in the detector nonlinearity
+                        function.  This will not ordinarily be accessed directly by users; instead,
+                        it will be accessed by the convenience function in this module that defines
+                        the nonlinearity function.
+
+    reciprocity_alpha - The normalization factor that determines the effect of reciprocity failure
+                        of the detectors for a given exposure time.  Typically, users would then
+                        simulate the reciprocity failure for a given image `im` and for the default
+                        WFIRST exposure time using
+
+        >>>> im_recip = im.addReciprocityFailure(galsim.wfirst.exptime,
+                                                 galsim.wfirst.reciprocity_alpha)
```

Comment:
Oh, and I'd recommend having such functions take `exp_time` and other _survey_ parameters (as opposed to parameters about the telescope or detectors) as optional keyword arguments, since investigating what the optimal choices are for these parameters is probably one application of this module.  The telescope and detector characteristics on the other hand are probably mostly set in stone at this point.